### ISSUE TITLE
Realtime Socket set to Null after Sign-In

### ISSUE THREAD

**Issue by Kuffs2205** [supabase-community/supabase-csharp#107]
As the title suggests:

Works fine in 0.12.2 but Client.Realtime.Socket is set to null after signing in when using 0.13.0 or 0.13.1 
( I only install the Supabase-csharp package not each individual library)

```
' client.Realtime.Session Not Null
  Dim session = Await client.Auth.SignIn(email, password)
' client.Realtime.Session is now Null
```

Also, Can I ask what version this is resolved in:
https://github.com/supabase-community/gotrue-csharp/pull/74
I have such trouble trying to work out if a fix is released or not. Are there any release notes?

**Comment by acupofjose** [comment]
[#74](https://github.com/supabase-community/gotrue-csharp/pull/74) was released on `0.13.0` you can see the release notes in the [CHANGELOG](https://github.com/supabase-community/supabase-csharp/blob/master/CHANGELOG.md#0130---2023-08-26)!

I'll need to look into the realtime socket being null, let me get back to you on that one.

**Comment by acupofjose** [comment]
@Kuffs2205 are you calling `client.InitializeAsync()` after it's been constructed? Or is it possible you're using a client that hasn't called that? That's the only way I can get your issue to duplicate currently.

**Comment by Kuffs2205** [comment]
As far as I know yes.

Am I doing something wrong here ?

https://www.screencast.com/t/wBzSsmCus

**Comment by acupofjose** [comment]
Very odd. No that looks fine?

It looks like you're hitting this line somehow:
https://github.com/supabase-community/supabase-csharp/blob/a8e16622591f67770dbbc14e11a52d32fd5589ff/Supabase/Client.cs#L225-L229


Which in turn references this:

https://github.com/supabase-community/realtime-csharp/blob/853c06c6b18e078cafe621bf0ada9e8cb01fee75/Realtime/Client.cs#L286-L298

So, it looks like the `SignedOut` event is being called without a `SignedIn` being followed up. 

Any thoughts @wiverson?

**Comment by Kuffs2205** [comment]
I just noticed that you released  0.13.3.
Tried it. No issue.

Went back to 0.13.2. Null Socket again.
Not sure about the sign out.  I seem to have a valid token back.

Anyway, it works with 13.3 so I'll all good. 
Thanks.

**Comment by acupofjose** [comment]
Yeah it was an incorrectly called disconnection on one of the event handlers! Glad it’s working for you. Maybe I can get a link for the changelog to show up on the wiki. Happy Friday!
